engine/src/main/java/org/destinationsol/game/projectile/PointProjectileBody.java

```diff
@@ -107,15 +109,22 @@ private ProjectileRayBack(Projectile projectile, SolGame game) {
 
         @Override
         public float reportRayFixture(Fixture fixture, Vector2 point, Vector2 normal, float fraction) {
-            if (fixture.getBody().getUserData() instanceof SolObject) {
-                SolObject o = (SolObject) fixture.getBody().getUserData();
-                boolean oIsMassless = o instanceof Projectile && ((Projectile) o).isMassless();
-                if (!oIsMassless && projectile.shouldCollide(o, fixture, game.getFactionMan())) {
-                    position.set(point);
-                    projectile.setObstacle(o, game);
-                    return 0;
-                }
+
+            //TODO This is a patch to smooth over contact between an Entity and a Projectile. Once Projectile has been
+            // converted to be an Entity, this can be removed.
+            Object userData = fixture.getBody().getUserData();
+            if (userData instanceof EntityRef) {
+                userData = new SolObjectEntityWrapper((EntityRef) userData);
+            }
+
+            SolObject o = (SolObject) userData;
```

Comment:
If `userData` is still not a `SolObject` at this point we should break / return -1

Reply:
`userData` will always be a `SolObject` by this point. The previous code assumed it would always be a SolObject, and my changes added the possibility of it being an EntityRef. I'll put in a comment that explains that